### PR TITLE
e2e test - Data Columns Names

### PR DESCRIPTION
**Pull request by rodrigosf672**
### QA Notes

This PR creates `data-columns.test.ts` test suite in order to verify data column names can be consistently generated in Positron's Data Explorer, via three methods:
- Opening CSV file with Python session open
- Opening CSV file with R session open
- Creating a data frame using Python script in the console

It covers most cases, including edge cases, and addresses #5915 .

@:win @:web @:data-explorer

### CODE REVIEW

File: test/e2e/tests/data-explorer/data-columns.test.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+		await app.workbench.dataExplorer.addFilterButton.click();
+		await app.workbench.dataExplorer.selectColumnButton.click();
+		await app.workbench.dataExplorer.verifyColumnHeaders(expectedColumnNames);
+	});
+});
+
+// R is not being included here due to `check_names` default R function.
+// Although `check_names` can be set to false, it makes things very complicated with column names with quotes, among other issues.
+test.describe('Verify data columns using data frame through console (with Python only)', () => {
+	test.afterEach(async ({ app, page }) => {
```
Comment (testlabauto) [inline test/e2e/tests/data-explorer/data-columns.test.ts:25]: I don't think you need `page`

File: test/e2e/tests/data-explorer/data-columns.test.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+import { test, tags, expect } from '../_test.setup';
+import { expectedColumnNames, pyColumnComparison } from './helpers/expected_columns';
+
+test.use({
+	suiteId: __filename
+});
+
+// Note that expecting modal popup (filter columns) to be out helps ensure nothing weird happens after double-escaping.
+test.describe('Verify data columns using UI', () => {
```
Comment (testlabauto) [inline test/e2e/tests/data-explorer/data-columns.test.ts]: I think you forgot to tag WEB & WIN.

File: test/e2e/tests/data-explorer/data-columns.test.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+		await openDataFile('data-files/data_explorer/data_columns.csv');
+		await app.workbench.dataExplorer.addFilterButton.click();
+		await app.workbench.dataExplorer.selectColumnButton.click();
+		await app.workbench.dataExplorer.verifyColumnHeaders(expectedColumnNames);
+	});
+});
+
+// R is not being included here due to `check_names` default R function.
+// Although `check_names` can be set to false, it makes things very complicated with column names with quotes, among other issues.
+test.describe('Verify data columns using data frame through console (with Python only)', () => {
```
Comment (testlabauto) [inline test/e2e/tests/data-explorer/data-columns.test.ts]: I think you forgot to tag WEB and WIN.

File: test/e2e/tests/data-explorer/data-columns.test.ts (hunk 2 above)
Comment (testlabauto) [reply test/e2e/tests/data-explorer/data-columns.test.ts]: oh, I see, the cases are tagged

**Review by jonvanausdeln — COMMENTED** [review]
I just tried locally on windows and all the tests fail.  Here is an example of the failures I see:

```
    Error: expect(received).toEqual(expected) // deep equality

    - Expected  - 0
    + Received  + 8

    @@ -6,10 +6,15 @@
        "column04",
        "123numeric_start",
        "!@#symbols",
        "中文字符",
        "naïve_column",
    +   "name,with,comma",
    +   "\"quoted\"",
    +   "multiline header",
    +   "supercalifragilisticexpialidocious_column_name_that_is_really_really_long_to_test_limits",
    +   "whitespace (tab)",
        "normal_name",
        "leading_space",
        "trailing_space",
        "both",
        "column04",
    @@ -29,6 +34,9 @@
        "Nome_Português",
        "Número_do_Pedido",
        "اسم_عربي",
        "رمز_المنتج",
        "שם_עברי",
    +   "מספר_פריט",
    +   "Heizölrückstoßabdämpfung",
    +   "100.000 pro Bevölkerung",
      ]

```

File: test/e2e/tests/data-explorer/data-columns.test.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+		await app.workbench.sessions.deleteAll();
+	});
+
+	test('Python can read CSV and verify column names', async ({ app, python }) => {
+		await app.workbench.console.pasteCodeToConsole(pyColumnComparison);
+		await app.workbench.console.sendEnterKey();
+		await app.workbench.console.waitForConsoleContents('True');
+	});
+});
+*/
```
Comment (midleman) [inline test/e2e/tests/data-explorer/data-columns.test.ts]: I'd highly recommend cleaning up / removing this commented-out code. It helps keep things tidy and easier to follow for whoever’s in here next. 😊

File: test/e2e/tests/data-explorer/helpers/expected_columns.ts [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+	'שם_עברי',
+	'מספר_פריט',
+	'Heizölrückstoßabdämpfung',
+	'100.000 pro Bevölkerung'
+];
+actual = [col.strip() for col in df.columns];
+print(actual == expected)
+`;
+
+*/
```
Comment (midleman) [inline test/e2e/tests/data-explorer/helpers/expected_columns.ts]: Same note here - commented-out blocks add a bit of mental overhead. Keeping tests clean makes it way easier for the next person jumping in.

File: test/e2e/tests/data-explorer/data-columns.test.ts [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+ *--------------------------------------------------------------------------------------------*/
+
+import { test, tags, expect } from '../_test.setup';
+import { expectedColumnNames } from './helpers/expected_columns';
+
+test.use({
+	suiteId: __filename
+});
+
+test.describe('Verify data columns using UI', { tag: [tags.WEB, tags.WIN, tags.DATA_EXPLORER] }, () => {
```
Comment (midleman) [inline test/e2e/tests/data-explorer/data-columns.test.ts]: Nit: our base test titles are usually the feature category/area. In this case something like: `Data Explorer: Columns`, `Data Explorer: Data Columns`, etc

File: test/e2e/tests/data-explorer/data-columns.test.ts [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+	test.afterEach(async ({ app, page }) => {
+		await page.getByRole('button', { name: 'Select Column' }).focus();
+		await page.keyboard.press('Escape');
+		await page.keyboard.press('Escape');
+		await expect(page.locator('.positron-modal-popup')).toHaveCount(0);
+		await app.workbench.console.clearButton.click();
+		await app.workbench.sessions.deleteAll();
+	});
```
Comment (midleman) [inline test/e2e/tests/data-explorer/data-columns.test.ts]: Not totally sure what this chunk is doing - is it meant as general cleanup? I don’t see any other tests in this spec yet, and adding extra test steps can sometimes introduce flakiness. Unless it’s needed for other tests in the file, I’d recommend leaving cleanup out for now.

**Review by midleman — COMMENTED** [review]
Left some small comments. Thanks for adding this test!

File: test/e2e/tests/data-explorer/data-columns.test.ts (hunk 7 above)
Comment (rodrigosf672) [reply test/e2e/tests/data-explorer/data-columns.test.ts]: It was a necessary clean up that I had done for when I had other tests. I'm putting it commented out in the end, in case anyone needs to pick that up later, and wants to add additonal tests to this suite.
